static/js/subs.js

```diff
@@ -3,10 +3,16 @@ var subs = (function () {
 var exports = {};
 
 function add_admin_options(sub) {
+    if (sub.subscribed == true) {
+        var can_moderate = _.contains(sub.permissions, 'can_moderate');
+        var can_moderate_stream = page_params.is_admin || can_moderate;
+        var can_add_remove_users = _.contains(sub.permissions, 'can_add_remove_users');
+    }
     return _.extend(sub, {
-        'is_admin': page_params.is_admin,
-        'can_make_public': page_params.is_admin && sub.invite_only && stream_data.is_subscribed(sub.name),
-        'can_make_private': page_params.is_admin && !sub.invite_only
+        'is_admin': can_moderate_stream,
```

Comment:
This code is hit even when `sub.subcribed` is false, in which case `can_moderate_stream` is undefined